cpu/esp_common/include/log/log_module.h

```diff
@@ -26,15 +26,6 @@ extern "C" {
 #include <stdio.h>
 #include "esp_common_log.h"
 
-#ifdef MODULE_LOG_PRINTFNOFORMAT
```

Comment:
What is the reason for changing the approach? I don't see any advantage to handle it as pseudomodule. ESPs require their own log implementation for compatibility and consistency reasons with the SDK and SDK libraries. It's a bit tricky but it works. Removing this part breaks the compatibility with module `log_printnoformat` so that `tests/log_printfnoformat` isn't compilable anymore.

Reply:
I don't think the current solution is good. The implementation of `log_printfnoformat` is being duplicated here for no reason, only to always include this header file. The way log backends are provided is explicitly defined on its documentation:
https://github.com/RIOT-OS/RIOT/blob/44ebc38caca3ce24914a6b543fb34703a304e660/core/include/log.h#L20-L27

> Removing this part breaks the compatibility with module log_printnoformat so that tests/log_printfnoformat isn't compilable anymore.

What do you mean by breaks compatibility?

In my opinion, if ESP needs a special logging which, for some reason, is not compatible with the existing ones, then one should not be able to use them. Another way is to provide their ESP version (e.g. `log_esp_prinfnoformat` or so), but always using modules, otherwise all this is really hard to see and maintain.

Reply:
> What do you mean by breaks compatibility?

With your changes, `tests/log_printfnoformat` is no longer compilable. This means that we either blacklist ESP boards for this test or define a dependency that we do not use the `esp_common/log` module if the `log_printfnoformat` module is used.

My guess is that the `log_printfnoformat` module exists to allow the user to disable `printf` logging to reduce ROM/RAM size. I think we should keep this option for ESP boards as well, even if the SDK library outputs are then no longer consistent and contain `printf` formatting sequences.

Reply:
> we do not use the esp_common/log module if the log_printfnoformat module is used.

I'm thinking we can do this using `Makefile.default`, right? ESP always tries to enable that logging backend by default and when specifically configured it is disabled.

Reply:
If this work, I'm fine with it.

Reply:
- A sanity check for dependencies is added in b9bea0b8b2749c18fb219d4bb50f909642e58bfc.
- The default is set in fd2255f45d3df7e6e25a90fb66fe18e8e26f4107

Reply:
Works as expected :+1: